kernel/vmemitcodes.ml

```diff
@@ -54,13 +54,74 @@ module RelocTable = Hashtbl.Make(struct
   let hash = hash_reloc_info
 end)
 
+module Positions :
+sig
+  type t
+  val of_list : int list -> t
+  val iter : (int -> unit) -> t -> unit
+end =
+struct
+
+type t = string
+(* Represent an ordered set of 32-bit integers as an array of successive diffs.
+   We use furthermore an approach where smaller integers use less bytes. Numbers
+   smaller than 255 are stored into one byte. This is a cheap way to compact
+   this data. *)
+
+let output buf n =
+  if n < 0xFF then Buffer.add_uint8 buf n
+  else
+    let () = Buffer.add_uint8 buf 0xFF in
+    Buffer.add_int32_be buf (Int32.of_int n)
+
+let input s pos =
+  let c = Char.code s.[!pos] in
+  if c < 0xFF then
+    let () = pos := !pos + 1 in
+    c
+  else
+    let c1 = Char.code s.[!pos + 1] in
+    let c2 = Char.code s.[!pos + 2] in
+    let c3 = Char.code s.[!pos + 3] in
+    let c4 = Char.code s.[!pos + 4] in
+    let () = pos := !pos + 5 in
+    (c1 lsl 24) lor (c2 lsl 16) lor (c3 lsl 8) lor c4
```

Comment:
I prefer having an explicitly purely functional type, because this is stored in the vo file. I'm willing to leave a comment for the future OCaml bump though.